proposals/NNNN-transferring-parameters-and-results.md

```diff
@@ -0,0 +1,320 @@
+# `transferring` parameter and result types
+
+* Proposal: [SE-NNNN](NNNN-transferring-parameteres-and-results.md)
+* Authors: [Michael Gottesman](https://github.com/gottesmm), [Holly Borla](https://github.com/hborla), [John McCall](https://github.com/rjmccall)
+* Review Manager: TBD
+* Status: **Awaiting review**
+* Implementation: On `main` gated behind `-enable-experimental-feature TransferringArgsAndResults`
+* Previous Proposal: [SE-0414: Region-based isolation](/proposals/0414-region-based-isolation.md)
+* Review: ([pitch](https://forums.swift.org/...))
+
+
+## Introduction
+
+This proposal extends region isolation to enable an explicit `transferring`
+annotation to denote when a parameter or result value is required to be in a
+disconnected region, allowing the callee or the caller, respectively, to
```

Comment:
I am a little worried that this is going to sound to people like these are disconnected parameters (which is an extension as we have spoken about). I think it would be clearer if you included verbiage that says that the values need to be disconnected upon function start in the callee and function return in the caller. Then it is really clear that we are allowing for the transferring values to be merged into other regions. That being said, I am willing to be convinced that this is too much detail here.

Reply:
Updated to say the "disconnected region" requirement only applies at the function boundary